crates/wysiwyg/src/tests/test_mentions.rs

```diff
@@ -594,6 +594,73 @@ fn can_insert_at_room_mention() {
     assert_eq!(tx(&model), "<a style=\"some css\" data-mention-type=\"at-room\" href=\"#\" contenteditable=\"false\">@room</a>&nbsp;|")
 }
 
+#[test]
+fn get_mentions_state_for_no_mentions() {
+    let model = cm("<p>hello!|</p>");
+    assert_eq!(model.get_mentions_state(), MentionsState::default())
+}
+
+#[test]
+fn get_mentions_state_for_user_mention() {
+    let model = cm("<p>hello <a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a>!|</p>");
+    let mut state = MentionsState::default();
+    state.user_ids.insert("@alice:matrix.org".into());
+    assert_eq!(model.get_mentions_state(), state)
+}
+
+#[test]
+fn get_mentions_state_for_multiple_user_mentions() {
+    let model = cm("<p>hello <a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a> and <a data-mention-type=\"user\" href=\"https://matrix.to/#/@bob:matrix.org\" contenteditable=\"false\">Bob</a>!|</p>");
+    let mut state = MentionsState::default();
+    state.user_ids.insert("@alice:matrix.org".into());
+    state.user_ids.insert("@bob:matrix.org".into());
+    assert_eq!(model.get_mentions_state(), state)
+}
+
+#[test]
+fn get_mentions_state_for_at_room_mention() {
+    let model = cm("<p>hello <a data-mention-type=\"at-room\" href=\"#\" contenteditable=\"false\">@room</a>|");
+    let state = MentionsState {
+        user_ids: Default::default(),
+        has_at_room_mention: true,
+    };
+    assert_eq!(model.get_mentions_state(), state)
+}
+
+#[test]
+fn get_mentions_state_for_at_room_plain_mention() {
+    let model = cm("<p>hello @room|");
+    let state = MentionsState {
+        user_ids: Default::default(),
+        has_at_room_mention: true,
+    };
+    assert_eq!(model.get_mentions_state(), state)
+}
+
+#[test]
+fn get_mentions_state_for_multiple_user_and_at_room_mentions() {
```

Comment:
Should we add one test for content containing multiple `@room` mentions and duplicate user mentions?